consensus/istanbul/validator/default.go

```diff
@@ -28,12 +28,27 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/consensus/istanbul"
 	blscrypto "github.com/ethereum/go-ethereum/crypto/bls"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
 type defaultValidator struct {
 	address      common.Address
 	blsPublicKey blscrypto.SerializedPublicKey
+	uncompressed []byte
+}
+
+func newValidatorFromDataWithCache(data *istanbul.ValidatorDataWithCache) *defaultValidator {
+	uncompressed := data.Uncompressed
+	if len(uncompressed) == 0 {
+		log.Warn("no uncompressed")
+		uncompressed = nil
```

Comment:
Maybe we should uncompress here directly? It should be rare after some time, right?